tests/TestCase/Database/QueryTest.php

```diff
@@ -4460,6 +4460,79 @@ public function testGetValueBinder()
         $this->assertInstanceOf('\Cake\Database\ValueBinder', $query->getValueBinder());
     }
 
+    /**
+     * Test automatic results casting
+     *
+     * @return void
+     */
+    public function testCastResults()
+    {
+        $this->loadFixtures('Profiles');
+        $query = new Query($this->connection);
+        $fields = [
+            'id' => 'integer',
+            'user_id' => 'integer',
+            'is_active' => 'boolean'
+        ];
+        $typeMap = new TypeMap($fields + ['a' => 'integer']);
+        $results = $query
+            ->select(array_keys($fields))
+            ->select(['a' => 'id'])
+            ->from('profiles')
+            ->setSelectTypeMap($typeMap)
+            ->limit(1)
+            ->execute()
+            ->fetchAll('assoc');
+        $this->assertSame([['id' => 1, 'user_id' => 1, 'is_active' => false, 'a' => 1]], $results);
```

Comment:
@lorenzo 
this expects
```php
       'id' => 0
        'user_id' => 1
        'is_active' => false
        'a' => 0
```
on windows 
https://ci.appveyor.com/project/cakephp/cakephp/build/1.0.9754#L297

Reply:
add an order() clause

Reply:
actually it is odd that the is is 0, the fixture seems to be wrong

Reply:
see next test please (`testCastResultsDisable`)
when calls  `disableResultsCasting()`
`'a' => '1'` is true

Reply:
@saeideng What should I be looking for in that test?